### ISSUE TITLE
Allow non-string runConfig values

### ISSUE THREAD

**Issue by laneb** [TresAmigosSD/SMV#1167]
The like the app config, the runConfig only accepts strings as config values. I would like to be able to parameterize a module so that expects e.g. a list of values in the runConfig. In order to work around this for my current project, I am dumping the object to json before putting it in the runConfig, and loading it after retrieving it from the runConfig. Ideally, SMV would either a. handle the jsonification itself or b. just allow non-string values on the runConfig. If you include a non-string value you get a Java introspection error.

**Comment by AliTajeldin** [comment]
@laneb can you give an example of a non-string value?  The run config was intended as a simple k/v parameter list.

**Comment by laneb** [comment]
Yeah, e.g. I have a list or tuple of values like a date range that I need to pass to my module. I want to be able to do
```
df("MyModule", runConfig={"date_range": ["2018-02-01", "2018-02-02"]}
```
or perhaps I need to apply different date_ranges to different modules
```
runConfig =  {
    "date_ranges":  {
        "MyModule": ["2018-02-01", "2018-02-02"],
        "SomeOtherModule": ["2018-02-02", "2018-02-03"]
    }
}
df("MyModule", runConfig=runConfig)
```

**Comment by AliTajeldin** [comment]
Thats a tough one because we also need to allow user to specify run config k/v from command line.  How would user specify the above range from the command line?
We could perhaps add a helper to get the config as a json.  Similar to how we get config value as int, string, etc.

**Comment by laneb** [comment]
If we set the requirement that any value to be passed in this way must be jsonifiable and the user is using `smv-run` then they can give a json string for the config value. In our case we generate the object dynamically in an entrypoint script, but that's probably not a typical use-case.

**Comment by laneb** [comment]
This is definitely a nice-to-have, non critical feature. The workaround of manually jsonifying and dejsonifying works well enough.

**Comment by ninjapapa** [comment]
Will the customer driver fulfill this use case?

**Comment by laneb** [comment]
I don't think so. I'm not sure if we'll run into this case again, though.

**Comment by ninjapapa** [comment]
Now with generic modules, can we create a module which result as the time range, and your module depends on that "conf" module?
